src/Pact/Types/Swagger.hs

```diff
@@ -69,6 +96,36 @@ fixedLength i =
   set maxLength (Just $ fromIntegral i) .
   set minLength (Just $ fromIntegral i)
 
+replaceOrModify :: (Eq a) => (a, a) -> (a -> a) -> a -> a
+replaceOrModify (target, replacement) modifier curr
+  | (target == curr)  = replacement
+  | otherwise         = modifier curr
+
```

Comment:
(Small non-issue) It's slightly difficult to see what's going on from this type signature. If we're going to keep it and reuse it, do you mind changing it to something like 

```haskell
replaceOrModify :: (Eq a) => a -> a -> (a -> b) -> b -> b
replaceOrModify target curr k d = bool (f curr) d $ target == curr
```